src/ImageSharp/Processing/Processors/Convolution/EdgeDetectorCompassProcessor.cs

```diff
@@ -104,7 +99,7 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
             // we need a clean copy for each pass to start from
             using (ImageFrame<TPixel> cleanCopy = source.Clone())
             {
-                new ConvolutionProcessor<TPixel>(kernels[0]).Apply(source, sourceRectangle, configuration);
+                new ConvolutionProcessor<TPixel>(kernels[0], true).Apply(source, sourceRectangle, configuration);
```

Comment:
If `preserveAlpha` is always true, I wonder if we need it actually.

Reply:
Only for this algorithm. Other convolvers might not want it. 2 pass blur for example does not.